CredentialProvider.Microsoft/CredentialProviders/Vsts/MSAL/MsalBearerTokenProvidersFactory.cs

```diff
@@ -12,22 +14,26 @@ internal class MsalBearerTokenProvidersFactory : IBearerTokenProvidersFactory
         private readonly ILogger logger;
         private readonly IMsalTokenProviderFactory msalTokenProviderFactory;
 
-        public MsalBearerTokenProvidersFactory(ILogger logger,IMsalTokenProviderFactory msalTokenProviderFactory)
+        public MsalBearerTokenProvidersFactory(ILogger logger, IMsalTokenProviderFactory msalTokenProviderFactory)
         {
             this.msalTokenProviderFactory = msalTokenProviderFactory;
             this.logger = logger;
         }
 
-        public IEnumerable<IBearerTokenProvider> Get(string authority)
+        public IEnumerable<IBearerTokenProvider> Get(Uri authority)
         {
-            IMsalTokenProvider msalTokenProvider = msalTokenProviderFactory.Get(authority, logger);
-            return new IBearerTokenProvider[]
+            var options = EnvUtil.MsalAllowBrokerEnabled()
+                ? new [] {true, false}
+                : new [] {false};
+
+            foreach(bool brokerEnabled in options)
             {
-                new MsalCacheBearerTokenProvider(msalTokenProvider),
-                new MsalWindowsIntegratedAuthBearerTokenProvider(msalTokenProvider),
-                new MsalUserInterfaceBearerTokenProvider(msalTokenProvider),
-                new MsalDeviceCodeFlowBearerTokenProvider(msalTokenProvider)
-            };
+                IMsalTokenProvider msalTokenProvider = msalTokenProviderFactory.Get(authority, brokerEnabled, logger);
+                yield return new MsalSilentBearerTokenProvider(msalTokenProvider);
```

Comment:
Does this also attempt to log in with the default OS account? It should work on Windows.

Reply:
Yes it does: https://github.com/microsoft/artifacts-credprovider/blob/9671beb3c3ed537e04678a9bf36ada2f4777667d/CredentialProvider.Microsoft/CredentialProviders/Vsts/MSAL/MsalTokenProvider.cs#L134

Currently used as a fallback if there are no usable cached accounts